index.js

```diff
@@ -23,6 +31,84 @@ module.exports = {
   },
 
   async updateDeps(options) {
+    let manifestPath = join(options.target, 'package.json');
+    let manifestBuffer = await readFile(manifestPath);
+    let manifest = JSON.parse(manifestBuffer.toString());
+
+    let existingDeps = [
+      ...Object.keys(manifest.dependencies || {}),
+      ...Object.keys(manifest.devDependencies || {}),
+    ];
+
+    let uninstallTask = this.taskFor('npm-uninstall');
+    const packages = [
+      'ember-fetch',
+      'broccoli-asset-rev',
+      'ember-cli-app-version',
+      'ember-cli-clean-css',
+      'ember-cli-dependency-checker',
+      'ember-cli-sri',
+      'ember-cli-terser',
+
+      // Upstream TypeScript blueprint is too conservative
+      // at the same time, ember-data has been what's holding it back
+      '@types/ember',
```

Comment:
shouldn't these only be removed if you've passed the typescript flag? every one of these will show an error in the terminal if they can't be removed so let's not try to remove things that we know won't be there

Reply:
they're only removed if the exist in the original package.json. the upstream package.json is the source of truth here.

>  every one of these will show an error in the terminal i

nay, there is a filter